lib/vrl/tests/tests/expressions/assignment/indexing.vrl

```diff
@@ -0,0 +1,20 @@
+# result:
+#
+# [
+#   [0, 1],
+#   [3, null],
+#   [5, 1, 4, 3, null, null, 6]
+# ]
+
+v1 = 0
```

Comment:
Interesting, so indexing into anything that is not an array will just discard the pervious value?

Reply:
It's not the indexing that replaces the previous value, it's the actual assignment.

`v1[0]` would return `null` (because `v1` is not an array, and we've made the decision to never error when querying).

`v1[0] = "foo"` will either (1) replace the first value of the array with `"foo"`, or if the value wasn't an array yet, convert it to one, and set the first value, yes.

Reply:
Nice, thanks for the background info 👍